File: native/Android/ui/SeasonalCalendarActivity.java
package com.example.produce.ui;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.example.produce.R;
import java.util.ArrayList;
import java.util.List;

public class SeasonalCalendarActivity extends AppCompatActivity {

    private TextView tvCurrentSeason;
    private ListView lvSeasonalCrops;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_seasonal_calendar);

        tvCurrentSeason = findViewById(R.id.tvCurrentSeason);
        lvSeasonalCrops = findViewById(R.id.lvSeasonalCrops);

        setupSeasonalData();
    }

    private void setupSeasonalData() {
        // 模擬當前節氣與當季農產品
        String currentSeason = "立秋 (約 8/7 - 8/22)";
        tvCurrentSeason.setText("目前節氣：" + currentSeason);

        List<String> crops = new ArrayList<>();
        crops.add("🍉 西瓜 - 挑選訣竅：拍打聲音清脆，蒂頭捲曲");
        crops.add("🥒 小黃瓜 - 挑選訣竅：瓜體直挺，表面有刺");
        crops.add("🍆 茄子 - 挑選訣竅：表皮光滑發亮，蒂頭無枯萎");
        crops.add("🥬 空心菜 - 挑選訣竅：葉片翠綠，莖部不發黑");

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, crops);
        lvSeasonalCrops.setAdapter(adapter);
    }
}

File: native/Android/ui/SeasonalCalendarActivity.kt
package com.example.produceapp.ui

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import com.example.produceapp.R
import com.example.produceapp.network.RetrofitClient
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class SeasonalCalendarActivity : AppCompatActivity() {

    private lateinit var tvCurrentSeason: TextView
    private lateinit var lvSeasonalCrops: ListView
    private lateinit var viewModel: ProduceViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_seasonal_calendar)

        tvCurrentSeason = findViewById(R.id.tvCurrentSeason)
        lvSeasonalCrops = findViewById(R.id.lvSeasonalCrops)

        val factory = ProduceViewModelFactory(RetrofitClient.instance)
        viewModel = ViewModelProvider(this, factory).get(ProduceViewModel::class.java)

        setupSeasonalData()
    }

    private fun setupSeasonalData() {
        lifecycleScope.launch {
            viewModel.seasonalCrops.collectLatest { crops ->
                if (crops.isNotEmpty()) {
                    // Assuming first crop's season is representative or we need a separate API for current season info
                    val currentSeason = crops.firstOrNull()?.season ?: "Unknown"
                    tvCurrentSeason.text = "目前節氣：$currentSeason"

                    val cropStrings = crops.map { "${it.cropName} - ${it.description}" }
                    val adapter = ArrayAdapter(this@SeasonalCalendarActivity, android.R.layout.simple_list_item_1, cropStrings)
                    lvSeasonalCrops.adapter = adapter
                }
            }
        }
    }
}
